### PR TITLE
Ssh keys

### PR DESCRIPTION
**Pull request by tpfau**
Added OS specific instructions for ssh key generation and setup.
I tested the instructions on aalto-linux and windows. For the latter, I could not test the ssh-add part  since I don't have a windows machine with Admin privileges, and that's necessary for the Openssh-agent. I will give it a shot at my private machine at home.
Windows stores the .ssh directory in the user profile. Interestingly, ssh does decipher an key location indicated as `~/.ssh/key_name` even on windows, so the final configuration file examples don't need to be changed.

### CODE REVIEW

**Comment by tpfau** [comment]
Needs some more fixes.

**Comment by tpfau** [comment]
Please don't merge yet until the ssh-add command is tested.

**Comment by tpfau** [comment]
Tested on Windows and works. So can be merged after review.

File: scicomp/ssh.rst
Comment (rkdarst) [inline scicomp/ssh.rst:90]: Normally I would recommend one key per client computer, for all hosts it would connect to.  What are your thoughts on that?

I can think of some reasons for multiple (e.g. on Triton I have one passwordless one for internal ssh and one with password for git).  And maybe being more controlled has less risk of being interefred by other things.

Also, in the comments, I would always include the source computer.  When you look at Triton and see all the old keys there, you want to be able to easily see what is still relevant and what is obsolete.  (at least for me default sometimes seems OK)

File: scicomp/ssh.rst
Comment (rkdarst) [inline scicomp/ssh.rst:118]:
```suggestion
   If you connect from outside aalto it is useful to first set up the config file :ref:`as described below <example_config_for_ssh>`. 
```

File: scicomp/ssh.rst
Comment (rkdarst) [inline scicomp/ssh.rst:166]: With the default key name, is it automatically found?

File: scicomp/ssh.rst
Comment (rkdarst) [inline scicomp/ssh.rst:176]: do Aalto ubuntus use some sort of Ubuntu-managed thing different from ssh-agent, or is that ssh-agent?  Hm, I should check myself...

**Review by rkdarst — APPROVED** [review]
This was very needed!

I have some comments below (at least one for me to check).  It's really interesting to think about this, since there are so many different things that go into it and are needed to make things simple but without things that will catch some people.

File: scicomp/ssh.rst
Comment (tpfau) [reply scicomp/ssh.rst:166]: I never tried this with ssh-add. Anyone else?

File: scicomp/ssh.rst
Comment (tpfau) [reply scicomp/ssh.rst:176]: They use ssh-agent, but I'm not sure, whether you can start it yourself.

File: scicomp/ssh.rst
Comment (rkdarst) [reply scicomp/ssh.rst:166]: ssh-add with no arguments uses the default key location.  I only give a key name where I have a separate key with password, separate from the default passwordless one.

**Pull request merged**
